modules/auxiliary/scanner/http/gitlab_login.rb

```diff
@@ -0,0 +1,92 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'metasploit/framework/credential_collection'
+require 'metasploit/framework/login_scanner/gitlab'
+
+class Metasploit3 < Msf::Auxiliary
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::AuthBrute
+
+  def initialize
+    super(
+      'Name'           => 'GitLab Login Utility',
+      'Description'    => 'This module attempts to login to a GitLab instance using a specific user/pass.',
+      'Author'         => [ 'Ben Campbell' ],
+      'License'        => MSF_LICENSE
+    )
+
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('USERNAME', [ true, 'The username to test', 'root' ]),
+        OptString.new('PASSWORD', [ true, 'The password to test', '5iveL!fe' ]),
+        OptString.new('TARGETURI', [true, 'The path to GitLab', '/'])
+      ], self.class)
+
+    register_autofilter_ports([ 80, 443 ])
+
+    deregister_options('RHOST')
+  end
+
+  def run_host(ip)
+    uri = normalize_uri(target_uri.path.to_s, 'users', 'sign_in')
+    res = send_request_cgi(
```

Comment:
You can actually override #check_setup and move this portion to your Metasploit::Framework::LoginScanner::GitLab. Maybe next time.

About #check_setup:
https://github.com/rapid7/metasploit-framework/blob/master/lib/metasploit/framework/login_scanner/http.rb#L164